tools/perf/test.js

```diff
@@ -47,47 +47,63 @@ function color(str, col) {
   return str;
 }
 
-function runAll(sample, callback) {
-  var results = {};
-  var times = {};
-  // run is recursive and thus sequential so that node doesn't spawn all the 
-  // processes at once
-  run();
-
-  function run() {
-    exec(command, { cwd: __dirname }, function (error, stdout, stderr) {
-      if (!program.raw && program.progress) {
-        process.stdout.write(color('.', INDICATOR));
-      }
-      if (error) {
-        console.log(error.toString());
-      }
-      var data = JSON.parse(stdout);
-      for (var scenario in data) {
-        if (!times[scenario]) {
-          times[scenario] = [];
-        }
-        times[scenario].push(data[scenario]);
-      }
-      if (times[scenario].length !== sample) {
-        run();
-      } else {
-        for (scenario in times) {
-          var mean = util.mean(times[scenario]);
-          results[scenario] = {
-            mean: mean,
-            stdev: util.stdev(times[scenario], mean),
-            sample: sample
-          };
-        }
-        callback(results);
-      }
-    });
+runAll(parseInt(program.sampleSize)).then(printResults);
+
+async function runAll(sampleSize) {
+  const results = {};
+
+  const testData = JSON.parse(fs.readFileSync(`${__dirname}/fixtures/benchmarks.json`).toString());
+  for (let sampleName in testData) {
```

Comment:
Would `benchmarkName` be a better name? We use `sample` in the statistical meaning in this file, too.